vowpalwabbit/core/src/reductions/gd.cc

```diff
@@ -1503,6 +1561,8 @@ std::shared_ptr<VW::LEARNER::learner> VW::reductions::gd_setup(VW::setup_base_i&
 
   all.weights.stride_shift(static_cast<uint32_t>(::ceil_log_2(stride - 1)));
 
+  resize_ppw_state(*g, ppw);
```

Comment:
make this logic part of the constructor in line 1428 - now that we have that info so there's no need to resize